kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/KafkaServiceReconciler.java

```diff
@@ -71,28 +75,51 @@ public List<EventSource<?, KafkaService>> prepareEventSources(EventSourceContext
                 Secret.class,
                 KafkaService.class)
                 .withName(SECRETS_EVENT_SOURCE_NAME)
-                .withPrimaryToSecondaryMapper((KafkaService cluster) -> ResourcesUtil.localRefAsResourceId(cluster, cluster.getSpec().getTls().getCertificateRef()))
-                .withSecondaryToPrimaryMapper(secret -> ResourcesUtil.findReferrers(context,
-                        secret,
-                        KafkaService.class,
-                        service -> service.getSpec().getTls().getCertificateRef()))
+                .withPrimaryToSecondaryMapper(kafkaServiceToSecret())
+                .withSecondaryToPrimaryMapper(secretToKafkaService(context))
                 .build();
         InformerEventSourceConfiguration<ConfigMap> serviceToConfigMap = InformerEventSourceConfiguration.from(
                 ConfigMap.class,
                 KafkaService.class)
                 .withName(CONFIG_MAPS_EVENT_SOURCE_NAME)
-                .withPrimaryToSecondaryMapper(
-                        (KafkaService cluster) -> ResourcesUtil.localRefAsResourceId(cluster, asRef(cluster.getSpec().getTls().getTrustAnchorRef())))
-                .withSecondaryToPrimaryMapper(configMap -> ResourcesUtil.findReferrers(context,
-                        configMap,
-                        KafkaService.class,
-                        service -> asRef(service.getSpec().getTls().getTrustAnchorRef())))
+                .withPrimaryToSecondaryMapper(kafkaServiceToConfigMap())
+                .withSecondaryToPrimaryMapper(configMapToKafkaService(context))
                 .build();
         return List.of(
                 new InformerEventSource<>(serviceToSecret, context),
                 new InformerEventSource<>(serviceToConfigMap, context));
     }
 
+    @VisibleForTesting
```

Comment:
This is the same approach that @robobario took in the KafkaProxyReconcilerTests.  I think it is useful pattern.

Reply:
Yes, agreed. I think that might be a better pattern.